### PR TITLE
Add atomic64 type traits

### PR DESCRIPTION
**Pull request by yuriykoch**
Provides has_atomic_support and bits_eq type traits to make possible separate code path for the atomic64 types at the compile time.

This PR depends on:
#77 - 5805017 - util/type_traits.h was added

Please, review changes starting from b1af951 - "Add has_atomic_support and bits_eq type traits"

### CODE REVIEW

File: util/type_traits.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                     contains<T, tailT...>>::type {};
+
+/**
+ * @brief Verify type has the given number of bits
+ */
+template <typename T, size_t bits>
+struct bits_eq :
+    std::conditional<(sizeof(T) * CHAR_BIT) == bits,
+                     std::true_type,
+                     std::false_type>::type {};
```
Comment (ProGTX) [inline util/type_traits.h:40]:
```suggestion
template <typename T, size_t bits>
using bits_eq = std::bool_constant<(sizeof(T) * CHAR_BIT) == bits>;
```

Prefer writing type traits in terms of aliases wherever possible - it's clearer and helps with compilation times.

File: util/type_traits.h [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    contains<T,
+             int, unsigned int,
+             long, unsigned long,
+             long long, unsigned long long,
+             float>;
+
+/**
+ * @brief Checks whether T is a floating-point sycl type
+ */
+template <typename T> struct is_cl_float_type {
```
Comment (ProGTX) [inline util/type_traits.h:59]: This could be `bool_constant` as well

**Review by ProGTX — APPROVED** [review]

File: util/type_traits.h (hunk 1 above)
Comment (yuriykoch) [reply util/type_traits.h]: Great catch, thanks

File: util/type_traits.h (hunk 2 above)
Comment (yuriykoch) [reply util/type_traits.h]: Fixed

**Comment by yuriykoch** [comment]
All dependencies are merged. PR contains only relevant changes now.

**Pull request merged**